contracts/resolvers/profiles/ExtendedDNSResolver.sol

```diff
@@ -2,18 +2,65 @@
 pragma solidity ^0.8.4;
 
 import "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
 import "../../resolvers/profiles/IExtendedDNSResolver.sol";
 import "../../resolvers/profiles/IAddressResolver.sol";
 import "../../resolvers/profiles/IAddrResolver.sol";
+import "../../resolvers/profiles/ITextResolver.sol";
 import "../../utils/HexUtils.sol";
+import "../../dnssec-oracle/BytesUtils.sol";
 
+/**
+ * @dev Resolves names on ENS by interpreting record data stored in a DNS TXT record.
+ *      This resolver implements the IExtendedDNSResolver interface, meaning that when
+ *      a DNS name specifies it as the resolver via a TXT record, this resolver's
+ *      resolve() method is invoked, and is passed any additional information from that
+ *      text record. This resolver implements a simple text parser allowing a variety
+ *      of records to be specified in text, which will then be used to resolve the name
+ *      in ENS.
+ *
+ *      To use this, set a TXT record on your DNS name in the following format:
+ *          ENS1 <address or name of ExtendedDNSResolver> <record data>
+ *
+ *      For example:
+ *          ENS1 2.dnsname.ens.eth a[60]=0x1234...
+ *
+ *      The record data consists of a series of key=value pairs, separated by spaces. Keys
+ *      may have an optional argument in square brackets, and values may be either unquoted
+ *       - in which case they may not contain spaces - or single-quoted. Single quotes in
+ *      a quoted value may be backslash-escaped.
+ *
+ *      Record types:
```

Comment:
How about contenthash?

Reply:
There's no browsers that would prioritize a contenthash in ENS over the A records of the domain right now, and no reasonable prospect that that'll change any time soon.

Reply:
true, browsers won't care about any of these records, but [some people still want to associate IPFS with DNS names](https://dnslink.dev/). it would be easy to implement because it doesn't require a parameter, (`c=<hash>`), and then you have complete resolver support. 

ps. I know these records need to be evm-readable, but you have 255 chars to work with and humans involved. how about `addr[60]=0x00`, `text[avatar]='http...'`, and `contenthash=<hash>`.  maybe support `addr=0x00` and `a=0x00` as well.


pps. `dnsname.ens.eth` is a mouthful, `dns.resolver` would be nice.